docs/design/jet/025-jet-job-member-pruning.md

```diff
@@ -0,0 +1,141 @@
+# Jet job member pruning
+
+|||
+|---|---|
+|Related Jira|[HZ-1605](https://hazelcast.atlassian.net/browse/HZ-1605)|
+|Document Status / Completeness|IN PROGRESS|
+|Requirement owner|Sandeep Akhouri|
+|Developer(s)|Sasha Syrotenko|
+|Quality Engineer|Isaac Sumner|
+|Technical Reviewers|TBD|
+|Version|5.4|
+
+## Background
+
+Before Hazelcast Platform 5.4, Jet job was always deployed to all data members. If some DAG vertex isn’t using all 
+members, it creates no-op processors on the rest, but the system still creates queues to/from those vertices
+and starts the processors, even though it completes immediately and the queues are closed with a `DONE_ITEM`. 
+If some member isn’t used at all, the DAG is still deployed to it, and the master has to send it to it and wait 
+for the completion. Even though the processors are no-op, or have no data to process, it’s an unnecessary overhead, 
+which becomes noticeable in very small batch jobs.
+
+## Terminology
+
+
+## Goals
+
+- deploy a job only on members with required partitions on them
+- 
+
+Non-goals are :
+- support member pruning for streaming jobs. 
```

Comment:
Some parts of streaming job might theoretically benefit from member pruning  as well (eg. if we supported hash join in streaming SQL jobs). Are you sure, that in our current streaming jobs (at least the ones created from SQL) it will never be used? It could happen in manually created streaming DAGs.

Reply:
>  Are you sure, that in our current streaming jobs (at least the ones created from SQL) it will never be used? 

Yes, at the moment I see no possibilities to use it at least for SQL jobs.